katdal/vis_flags_weights.py

```diff
@@ -276,6 +277,10 @@ class ChunkStoreVisFlagsWeights(VisFlagsWeights):
         Should be True if `corrprods` is `None`.
     van_vleck : {'off', 'autocorr'}, optional
         Type of Van Vleck (quantisation) correction to perform
+    index : tuple of slice, optional
+        Slice expression to apply to each array before combining them. At the
+        moment this can only have two elements (no slicing of baselines),
+        because ``weights_channel`` only has time and frequency dimensions.
```

Comment:
There would probably also be issues if, say, you sliced away the auto-correlations, since they'd then be missing when you tried to use them to compute weights.

Reply:
And the correlation products have a funny ordering, so a unit-step slice is not that useful (except, as you say, for distinguishing autocorrs and crosscorrs).